partner_second_lastname/models/res_partner.py

```diff
@@ -96,17 +97,18 @@ def _get_inverse_name(self, name, is_company=False):
                 result['lastname2'] = parts[1]
         return result
 
-    @api.one
+    @api.multi
     @api.constrains("firstname", "lastname", "lastname2")
     def _check_name(self):
         """Ensure at least one name is set."""
         try:
             super(ResPartner, self)._check_name()
         except exceptions.EmptyNamesError as error:
-            if not self.lastname2:
-                raise error
+            for partner in self:
+                if not partner.lastname2:
+                    raise error
 
-    @api.one
+    @api.multi
```

Comment:
It's good to remove the `@api.one` calls, just remember than in v10 `@api.multi` is the same as removing the decorator.